systemtest/src/test/java/io/strimzi/systemtest/KafkaClusterTest.java

```diff
@@ -253,4 +254,34 @@ public void testClusterWithCustomParameters() {
         assertThat(jsonString, valueOfCmEquals("KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR", "5"));
         assertThat(jsonString, valueOfCmEquals("KAFKA_TRANSACTION_STATE_LOG_REPLICATION_FACTOR", "5"));
     }
+    @Test
+    @Resources(value = "../examples/templates/cluster-controller", asAdmin = true)
+    @OpenShiftOnly
+    public void testDeployKafkaOnPersistentStorage() {
+        int expectedZKPods = 3;
+        int expectedKafkaPods = 3;
+        Oc oc = (Oc) this.kubeClient;
+        String clusterName = "openshift-my-cluster";
+        oc.newApp("strimzi-persistent", map("CLUSTER_NAME", clusterName));
+        oc.waitForStatefulSet(zookeeperStatefulSetName(clusterName), expectedZKPods);
+        oc.waitForStatefulSet(kafkaStatefulSetName(clusterName), expectedKafkaPods);
+
+        List<String> volumes = oc.list("pvc");
+        assertTrue(volumes.size() == (expectedZKPods + expectedKafkaPods));
+
+        //Checking Persistent volumes for Zookeeper nodes
+        for (int i = 0; i < expectedZKPods; i++) {
+            String volumeName = "data-" + zookeeperStatefulSetName(clusterName) + "-" + i;
```

Comment:
As before it's the name for a PersistentVolumeClaim not a PersistentVolume, so `volumeName' should be renamed imho.

Reply:
Good point from @ppatierno ... this kind of confused me so I thought this is somehow checking the existence of the PVs it self. (which might be useful, but probably hard to implement since the actual PV names can differ - they might not be dependent on the cluster name).

Reply:
We should factor out the generation of the volumeName into a method like we have with `zookeeperStatefulSetName()` for the SS.

Reply:
@tombentley we already had this discussion during one of my PR and we decided to have this one `getPersistentVolumeClaimName` in the `AbstractCluster` base class but it's not a static method as the others; the initial version was static then we decided to change it because that time being static was not necessary.

Reply:
@ppatierno it also relates the recent discussion about:

* whether the CC should be a dependency of the system tests (so it can access these methods), 
* whether the should just be duplicated in the system tests (which is what we current do)
* or whether we should create some minimal module (a dependency of both the CC and the system tests) to hold such constants and naming scheme methods (which seems like overkill, at least at the moment).

In any case we definitely should not go sprinkling these naming scheme expressions all through the code base, as it will make life harder should we ever need to change them.

Reply:
@tombentley I totally agree with you. Maybe for now @asorokhtey could get the code from the `getPersistentVolumeClaimName` method in `AbstractCluster` class and duplicate it here but we absolutely need a plan for doing what you are saying.